lib/proxy.js

```diff
@@ -133,6 +134,11 @@ var server = http.createServer(function(req, res) {
 
 
   function _proxy(target, cb) {
+
+    var config = configLoader.load();
+    var str = target.url;
+    target.url = str.replace('localhost',config.hostname);
```

Comment:
Just from looking, have not tested in our VM. Firstly, configLoader is async. We combine config options passed via command line, in the defaults.yaml, and in production we have an additional yaml file that overrides defaults. Just a long-winded way of saying `configLoader.load()` doesn't do what seems obvious at first.

It's going to be better to just convert the server method in this file to a function that accepts the config and returns the current server instance.

So from here: https://github.com/ProboCI/probo-proxy/blob/master/index.js#L13
will change to something like `var server = require('./lib/proxy').server(config)`

Since you passed the config, you can use it in `_proxy()`

Reply:
Having some issue with this. server is a variable and not a function so it isn't letting me pass in config. I'm trying to find another way to do this but not convinced I can do it right here.